hyrisecockpit/api/app/workload/controller.py

```
@@ -23,49 +23,40 @@ def get(self) -> List[Workload]:
         """Get all Workloads."""
         return WorkloadService.get_all()
 
-    @api.response(409, "A Workload with the given ID already exists.")
+    @api.response(409, "A Workload with the given folder name already exists.")
     @accepts(schema=WorkloadSchema, api=api)
     @responds(schema=WorkloadSchema, api=api)
     def post(self) -> Union[Workload, Response]:
         """Create a Workload."""
-        workload = WorkloadService.create(request.parsed_obj)
-        if workload is None:
-            return Response(status=409)
-        else:
-            return workload
+        interface: WorkloadInterface = request.parsed_obj
+        workload = WorkloadService.create(interface)
+        return Response(status=409) if workload is None else workload
```

Mabey we can just simply return the error code as an integer from the service. Then we can simply return it in the response and we don't need to check if the answer from the service is None.

---

That's definitely possible and might be more flexible in terms of adding new error events, however, we have not added error codes previously. On the other hand, checking for specific responses (i.e. `None`) and explicitly returning a predefined `Response` is easier to grasp and increases readability. Would you be okay with this approach?

---

It's just some stylistic choice, so I'm fine with it 🤓.